modules/fedimint-lnv2-tests/tests/tests.rs

```diff
@@ -288,3 +288,40 @@ async fn receive_operation_expires() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn rejects_wrong_network_invoice() -> anyhow::Result<()> {
+    let fixtures = fixtures();
+    let fed = fixtures.new_default_fed().await;
+    let gateway = gateway(&fixtures, &fed).await;
+    let gateway_api = gateway.versioned_api().clone();
+
+    // Signet invoice should fail on regtest
+    let signet_invoice = mock::signet_bolt_11_invoice();
```

Comment:
Please do not assign an extra variable here but call the constructor directly in send for consistency.